src/main/java/edu/byu/cs/util/RepoUrlValidator.java

```diff
@@ -0,0 +1,106 @@
+                return String.format("https://%s/%s/%s", domainName, githubUsername, repositoryName);
+            }
+        }
+
+        throw new InvalidRepoUrlException("Could not validate repo url given '" + repoUrl + "'.");
+    }
+
+    /**
+     * Generates an array of Strings representing Regexes for url matching purposes.
+     * The array allows for matches from any of multiple formats, and the first match will proceed as a success.
+     * The regexes will be evaluated in case-insensitive mode on a whitespace-trimmed string.
+     * <br>
+     * The regexes should be formed to extract the minimum amount of information while accepting the largest breadth
+     * of possible urls. This allows the URL to be constructed into a consistent format for uniqueness checking
+     * while also being easy for students to use; they should be able to submit any of the URLs from their repo.
+     * <br>
+     * The regex should contain exactly three <i>capturing groups</i> which will be interpreted as:
+     * <ol>
+     *     <li>The domain name of the link. This will be converted to lower case for consistency.</li>
+     *     <li>The username. Case sensitivity will be preserved.</li>
+     *     <li>The repo name. Case sensitivity will be preserved.</li>
+     * </ol>
+     * @return An array of strings representing regexes.
+     */
+    private String[] getRepoRegexPatterns() {
+        // NOTE: This is the place where we require GitHub links. The rest of the grading system behind this wall
+        // will function properly with any link that can be run with `git clone`, but we have arbitrarily chosen
+        // to require "github.com" links for now.
+        // NOTE: Acknowledging the current decision, this system has been set up to be domain name agnostic.
+        // Depending on how similarly another provider formats their links, it can be simple to accept BitBucket/GitLabs
+        // any other repo link in addition to GitHub links.
+
+        // Strictly match the beginning of the URL to ensure it's a github.com domain (and not fake-github.com)
+        // GitHub Username may only contain alphanumeric characters or single hyphens, and cannot begin or end with a hyphen.
+        // GitHub repository name can only contain ASCII letters, digits, and the characters ., -, and _. The .git extension will be removed.
+
+        // View the playground regex: https://regex101.com/r/r3dDAW/8
+
+        String domainName = "(github\\.com)";
+        String userName = "([a-zA-z0-9-]+)";
+        String repoName = "([\\w.-]+?)";
+        return new String[]{
+                "^(?:https?://)?(?:www\\.)?"+domainName+"/"+userName+"/"+repoName+"(?:\\.git|/|\\?|#|$)", // https
+                "^git@"+domainName+":"+userName+"/"+repoName+"\\.git$" // ssh
+        };
+    }
+
+    public static class InvalidRepoUrlException extends GradingException {
+        InvalidRepoUrlException(String message) {
+            super();
```

Comment:
This call to `super();` should pass in the parameter received in the constructor call. Robust linting should have warned about this unused variable.